receiver/receivercreator/README.md

```diff
@@ -438,3 +438,186 @@ service:
 
 The full list of settings exposed for this receiver are documented [here](./config.go)
 with detailed sample configurations [here](./testdata/config.yaml).
+
+
+## Generate receiver configurations from provided Hints
+
+Currently this feature is only supported for K8s environments and the `k8sobserver`.
+
+The feature for K8s is enabled with the following setting:
+
+```yaml
+receiver_creator/metrics:
+  watch_observers: [ k8s_observer ]
+  hints:
+    k8s:
+      enabled: true
+```
+
+Users can use the following annotations to automatically enable receivers to start collecting signals from the target Pods/containers.
+
+### Supported metrics annotations
+
+#### Enable/disable discovery
+
+`opentelemetry.io.discovery/enabled` (example: `"true"`)
+
+By default `"true"`.
+
+#### Define scraper
+
+`opentelemetry.io.discovery/scraper` (example: `"nginx"`)
+
+#### Enable specific signals
+
+`opentelemetry.io.discovery/signals` (example: `"metrics,logs"`)
+
+By default all signals are enabled.
+
+#### Define configuration
+
+`opentelemetry.io.discovery/config.`
+
+(example: ```opentelemetry.io.discovery/config.endpoint: "http://`endpoint`/nginx_status"```,
+
+For `config.endpoint` specifically, if not provided it defaults to
+```opentelemetry.io.discovery/config.endpoint: "`endpoint`"``` (as it comes from the ) which is
+in form of `pod_ip:container_port`.
+
+**Example:**
+
+```yaml
+# configure scraper behavior
+opentelemetry.io.discovery/signals: "metrics,logs"
+opentelemetry.io.discovery/config.endpoint: "http://`endpoint`/nginx_status"
+opentelemetry.io.discovery/config.collection_interval: '20s'
+opentelemetry.io.discovery/config.initial_delay: '20s'
+opentelemetry.io.discovery/config.read_buffer_size: '10'
+opentelemetry.io.discovery/config.xyz: 'abc'
+
+# nested configs
+opentelemetry.io/discovery/config.nested: |
+  zyz: foo
+  bar: baz
```

Comment:
To prevent reserving the `nested` keyword for components configuration, could we just use the config keyword instead:
```suggestion
opentelemetry.io/discovery/config: |
  zyz: foo
  bar: baz
```

Reply:
Hmm `.nested` here is not a fixed "hint"/"indicator" it's just a random example config's key. I can change it to sth else to avoid confusion :).

Reply:
Fixed as per discussions at https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34427.